### PR TITLE
Add comment reply FE

### PR DESCRIPTION
**Pull request by dayanamdr**

### CODE REVIEW

File: codewell-fe/src/Comment/View.js [hunk 1]
```diff
@@ -1,22 +1,69 @@
...
+  }
+
+  function action(index) {
+    const cancelButton = [
+      <span onClick={() => onCancelCommentReply(index)}>Cancel</span>
+    ]
+    const replyToButton = [
+      <span onClick={() => onAddCommentReply(index)}>Reply to</span>
+    ]
+    return showReplyBox[index] === 1? cancelButton : replyToButton;
```
Comment (andreisamoila74) [inline codewell-fe/src/Comment/View.js:55]: :( 
```suggestion
    return showReplyBox[index] === 1 ? cancelButton : replyToButton;
```

File: codewell-fe/src/Comment/View.js [hunk 2]
```diff
@@ -1,22 +1,69 @@
...
+  }
+
+  function children(index) {
```
Comment (andreisamoila74) [inline codewell-fe/src/Comment/View.js:58]: children who?

File: codewell-fe/src/Comment/View.js [hunk 3]
```diff
@@ -28,26 +75,29 @@ function ViewComments(props) {
   }, [onCommentAddSuccess]);
 
   return (
+    <>
     <Layout style={pageStyle}>
```
Comment (andreisamoila74) [inline codewell-fe/src/Comment/View.js:79]: indent this please, the whole code included in the <Layout />

File: codewell-fe/src/__tests__/Comment/View.test.js [hunk 4]
```diff
@@ -53,7 +52,7 @@ test('Displays comments', async () => {
   expect(first_comment).not.toContain('Second comment');
 
   // Check the values of the second comment
-  const second_comment = comments[1].textContent;
+  const second_comment = comments[2].textContent;
```
Comment (andreisamoila74) [inline codewell-fe/src/__tests__/Comment/View.test.js:55]: you also have to check the new content, right?

File: codewell-fe/src/__tests__/Comment/View.test.js [hunk 5]
```diff
@@ -98,7 +97,7 @@ test('Check comments after one comment was added', async () => {
   await wait(() => expect(global.fetch).toHaveBeenCalled());
 
   const comments = screen.getAllByRole('listitem');
-  expect(comments).toHaveLength(2);
+  expect(comments).toHaveLength(4);
```
Comment (andreisamoila74) [inline codewell-fe/src/__tests__/Comment/View.test.js:100]: same here and below

File: codewell-fe/src/Comment/View.js [hunk 6]
```diff
@@ -1,22 +1,69 @@
...
-import { List, Button, Card, Layout } from 'antd';
+import { List, Button, Layout, Comment } from 'antd';
 
 import AddComment from './Add.js';
 
 function ViewComments(props) {
   const [comments, setComments] = useState([]);
   const [onCommentAddSuccess, setOnCommentAddSuccess] = useState([]);
+  const [replyBox, setReplyBox] = useState([]);
+  const [showReplyBox, setShowReplyBox] = useState([]);
```
Comment (andreisamoila74) [inline codewell-fe/src/Comment/View.js:12]: why do we need both states?

**Review by andreisamoila74 — CHANGES_REQUESTED** [review]

File: codewell-fe/src/Comment/View.js [hunk 7]
```diff
@@ -1,22 +1,69 @@
 import React, { useEffect, useState } from 'react';
 
 import 'antd/dist/antd.css';
-import { List, Button, Card, Layout } from 'antd';
+import { List, Button, Layout, Comment } from 'antd';
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:4]: let's put those in alphabetical order

File: codewell-fe/src/Comment/View.js [hunk 8]
```diff
@@ -1,22 +1,69 @@
...
+    backgroundColor: 'white',
+    'whiteSpace': 'pre',
+    width: '100%'
+  };
 
   function updateComments(onCommentAddSuccess) {
     setOnCommentAddSuccess(onCommentAddSuccess);
   }
 
+  function onAddCommentReply(index) {
```
[the remark this answers is not in the document]
Comment (S7012MY) [reply codewell-fe/src/Comment/View.js:28]: comentLine, we don't do snake_case in js

File: codewell-fe/src/Comment/View.js [hunk 9]
```diff
@@ -1,22 +1,69 @@
...
+    var addReply = { ...replyBox };
+    addReply[index] = 'true';
+    setReplyBox(addReply);
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:31]: the naming here is not clear enough. I'd use something like isReplyBoxVisible

**Review by S7012MY — CHANGES_REQUESTED** [review]
Please review your code again and name your variables in a right way

File: codewell-fe/src/Comment/View.js [hunk 10]
```diff
@@ -1,22 +1,70 @@
...
+    backgroundColor: 'white',
+    'whiteSpace': 'pre',
+    width: '100%'
+  };
 
   function updateComments(onCommentAddSuccess) {
     setOnCommentAddSuccess(onCommentAddSuccess);
   }
 
+  function onAddCommentReply(commentLine) {
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:28]: commentIndex

File: codewell-fe/src/Comment/View.js [hunk 11]
```diff
@@ -1,22 +1,70 @@
...
 import 'antd/dist/antd.css';
-import { List, Button, Card, Layout } from 'antd';
+import { Button, Comment, Layout, List } from 'antd';
 
 import AddComment from './Add.js';
 
 function ViewComments(props) {
   const [comments, setComments] = useState([]);
   const [onCommentAddSuccess, setOnCommentAddSuccess] = useState([]);
+  const [isReplyBoxVisible, setIsReplyBoxVisible] = useState([]);
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:11]: please add a comment explaining what this is used for

File: codewell-fe/src/Comment/View.js [hunk 12]
```diff
@@ -1,22 +1,70 @@
...
+  const [isReplyButtonVisible, setIsReplyButtonVisible] = useState([]);
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:12]: same here

File: codewell-fe/src/Comment/View.js [hunk 13]
```diff
@@ -1,22 +1,70 @@
...
+    var addReplyBox = { ...isReplyBoxVisible };
+    addReplyBox[commentLine] = undefined;
+    setIsReplyBoxVisible(addReplyBox);
+
+    var showCancelButton = { ...isReplyButtonVisible };
+    showCancelButton[commentLine] = undefined;
+    setIsReplyButtonVisible(showCancelButton);
+  }
+
+  function button(commentLine) {
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:48]: naming is bad

**Review by S7012MY — CHANGES_REQUESTED** [review]
Cancel button should be left of the post button in the post comment component

File: codewell-fe/src/Comment/Add.js [hunk 14]
```diff
@@ -58,7 +63,8 @@ function AddComment(props) {
           <TextArea placeholder={t('Leave a comment')}
             onChange={onCommentChange}/>
         </Form.Item>
-        <Form.Item>
+        <Form.Item style={buttonStyle}>
+          {props.cancelButton}
```
Comment (S7012MY) [inline codewell-fe/src/Comment/Add.js:67]: I wouldn't pass a button as props, just the function for onCancel

Comment (S7012MY) [reply codewell-fe/src/Comment/Add.js:67]: Also the id should also be added

File: codewell-fe/src/Comment/View.js [hunk 15]
```diff
@@ -1,22 +1,65 @@
...
     setOnCommentAddSuccess(onCommentAddSuccess);
   }
 
+  function onAddCommentReply(commentIndex) {
+    var addReplyBox = { ...isReplyBoxVisible };
+    addReplyBox[commentIndex] = 'true';
+    setIsReplyBoxVisible(addReplyBox);
+  }
+
+  function onCancelCommentReply(commentIndex) {
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:33]: onAdd and onCancel could be merged in the same function togleReplyBoxVisibility

File: codewell-fe/src/Comment/View.js [hunk 16]
```diff
@@ -1,22 +1,65 @@
...
+
+  function cancelButton(commentIndex) {
+    return <Button onClick={() => onCancelCommentReply(commentIndex)}>
+      Cancel</Button>;
+  }
+
+  function replyButton(commentIndex) {
+    return [
+      <span onClick={() => onAddCommentReply(commentIndex)}>
+        Reply to
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:47]: translation

File: codewell-fe/src/Comment/View.js [hunk 17]
```diff
@@ -28,26 +71,29 @@ function ViewComments(props) {
...
+      <Layout style={pageStyle}>
+        <Layout style={containerStyle}>
+          <List
+            itemLayout="horizontal"
+            dataSource={comments}
+            renderItem={(comment, commentIndex) => (
+              <List.Item>
+                <Comment
+                  author={comment.user_id}
+                  datetime={comment.date_posted}
```
Comment (S7012MY) [inline codewell-fe/src/Comment/View.js:84]: alphabetical order

**Review by S7012MY — CHANGES_REQUESTED** [review]

**Pull request closed without merging**